app/uk/gov/hmrc/excisemovementcontrolsystemapi/services/NrsServiceNew.scala

```diff
@@ -76,11 +79,30 @@ class NrsServiceNew @Inject() (
   }
 
   def submitNrs(workItem: WorkItem[NrsSubmissionWorkItem])(implicit hc: HeaderCarrier): Future[Done] =
-    //needs to call connector, and handle marking the workitems. IN PROGRESS
-    for {
-      _ <- nrsConnectorNew.sendToNrs(workItem.item.payload, correlationIdService.generateCorrelationId())
-      _ <- nrsWorkItemRepository.complete(workItem.id, ProcessingStatus.Succeeded)
-    } yield Done
+    nrsConnectorNew
+      .sendToNrs(workItem.item.payload, correlationIdService.generateCorrelationId())
+      .flatMap { _ =>
+        nrsWorkItemRepository
+          .complete(workItem.id, Succeeded)
+          .map(_ => Done)
+      }
+      .recoverWith {
+        case ex: UnexpectedResponseException if is4xx(ex.status) =>
```

Comment:
I'm slightly concerned about this case being marked as permenantly failed. Might be me being over-cautious, but *in theory* any 4xx should be our fault, do we really want to never resubmit them? Then again multiple failures would also be lost so maybe this isn't an issue?

Maybe something to chat about post-review/pre-merge